src/completion.zig

```diff
@@ -0,0 +1,201 @@
+const std = @import("std");
+const stdx = @import("stdx.zig");
+const assert = std.debug.assert;
+
+const RingBuffer = @import("ring_buffer.zig").RingBufferType;
+const BoundedArray = stdx.BoundedArrayType;
+
+const completion_entries = 256;
```

Comment:
This constant seems high to me - isn't the maximum completions, at least currently, limited by the number of keywords? Does it make sense to set this exactly to `keywords.len`?

Reply:
yes, 256 is high. setting it to `completion_entries = keywords.len`